update_and_render.sh

```diff
@@ -2,26 +2,42 @@
 
 set -e
 
+LND_DIR="$GOPATH/src/github.com/lightningnetwork/lnd"
+EXPERIMENTAL_PACKAGES="${EXPERIMENTAL_PACKAGES:-signrpc walletrpc chainrpc invoicesrpc watchtowerrpc}"
+PROTO_DIR="${PROTO_DIR:-build/protos}"
+WS_ENABLED="${WS_ENABLED:-true}"
+GITHUB_URL="https://github.com/lightningnetwork/lnd"
+
+rm -rf $PROTO_DIR
+
 # We'll default to fetching the latest version of the RPC files. Otherwise, we'll use the commit hash provided.
 commit="master"
 [ -n "$1" ] && commit=$1
 
-curl -o rpc.proto -s https://raw.githubusercontent.com/lightningnetwork/lnd/$commit/lnrpc/rpc.proto
-curl -o rpc.swagger.json -s https://raw.githubusercontent.com/lightningnetwork/lnd/$commit/lnrpc/rpc.swagger.json
+# Update lncli to the respective commit.
+pushd $LND_DIR
+COMMIT=$(git rev-parse HEAD)
+if [ $commit != "nocompile" ]; then
+  git reset --hard HEAD
+  git pull
+  git checkout $commit
+  COMMIT=$(git rev-parse HEAD)
+  make clean && make && make install tags="$EXPERIMENTAL_PACKAGES"
+fi
+popd
 
-# Generate the rpc.json file from rpc.proto, so that render.py can parse it.
-protoc -I. -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis --doc_out=json,rpc.json:. rpc.proto
+# Copy over all proto and json files from the checked out lnd source directory.
+mkdir -p $PROTO_DIR
+rsync -a --prune-empty-dirs --include '*/' --include '*.proto' --include '*.json' --exclude '*' $LND_DIR/lnrpc/ $PROTO_DIR/
 
-# Update lncli to the respective commit.
-pushd $GOPATH/src/github.com/lightningnetwork/lnd
-git reset --hard HEAD
-git pull
-git checkout $commit
-make clean && make && make install
+pushd $PROTO_DIR
+protoc -I. -I$GOPATH/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
```

Comment:
Needed to add an additional include here: `-I/usr/local/include` when trying this out on Linux.

Reply:
Yes, thanks, added that.